dashboard/app/models/user.rb

```diff
@@ -310,6 +311,12 @@ def update_and_add_users_school_infos
     )
   end
 
+  def complete_school_info
+    if user_school_infos.count > 0 && !school_info.complete?
+      errors.add(:school_info_id, "cannot add new school id")
```

Comment:
Sorry for not asking this earlier but where do these errors go? Are they logged somewhere?

Reply:
@bethanyaconnor Good question.  I followed a similar pattern in this model:

https://github.com/code-dot-org/code-dot-org/blame/48ed4c6da7251c81a4c9452c99fa0db164b320f3/dashboard/app/models/school_info.rb#L172

Reply:
It's a Rails feature!  See [Active Record Validations: Custom methods](https://guides.rubyonrails.org/active_record_validations.html#custom-methods).

Reply:
Thanks Brad!